src/overlays/actors/ovl_En_Bal/z_en_bal.c

```diff
@@ -64,108 +149,1026 @@ static DamageTable D_80A63CE0 = {
+            case TINGLE_MAP_WOODFALL:
+                mapItemId = GI_TINGLE_MAP_WOODFALL;
+                break;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bal/func_80A622A0.s")
+            case TINGLE_MAP_SNOWHEAD:
+                mapItemId = GI_TINGLE_MAP_SNOWHEAD;
+                break;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bal/func_80A623CC.s")
+            case TINGLE_MAP_ROMANI_RANCH:
+                mapItemId = GI_TINGLE_MAP_ROMANI_RANCH;
+                break;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bal/func_80A624DC.s")
+            case TINGLE_MAP_GREAT_BAY:
+                mapItemId = GI_TINGLE_MAP_GREAT_BAY;
+                break;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bal/func_80A62818.s")
+            case TINGLE_MAP_STONE_TOWER:
+                mapItemId = GI_TINGLE_MAP_STONE_TOWER;
+                break;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bal/func_80A62830.s")
+            default:
+                mapItemId = GI_TINGLE_MAP_CLOCK_TOWN;
+                break;
+        }
+        Actor_OfferGetItem(&this->picto.actor, play, mapItemId, 500.0f, 100.0f);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bal/func_80A629C4.s")
+void EnBal_SetupThankYou(EnBal* this) {
+    this->actionFunc = EnBal_ThankYou;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bal/func_80A62AC4.s")
+void EnBal_ThankYou(EnBal* this, PlayState* play) {
+    Player* player = GET_PLAYER(play);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bal/func_80A62B30.s")
+    if (Actor_ProcessTalkRequest(&this->picto.actor, &play->state)) {
+        player->stateFlags1 &= ~PLAYER_STATE1_20;
+        Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimationInfo, TINGLE_ANIM_TALK_TWIST);
+        this->forceEyesShut = false;
+        this->eyeTexIndex = EN_BAL_EYETEX_OPEN;
+        Message_StartTextbox(play, 0x1D17, &this->picto.actor);
+        this->prevTextId = 0x1D17;
+        this->picto.actor.flags &= ~ACTOR_FLAG_10000;
+        EnBal_SetupTalk(this);
+    } else {
+        func_800B85E0(&this->picto.actor, play, 200.0f, PLAYER_IA_MINUS1);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bal/func_80A62B78.s")
+void EnBal_UpdateHead(EnBal* this, PlayState* play) {
+    Player* player;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bal/func_80A62CF0.s")
+    if (this->watchTarget == EN_BAL_WATCHTRG_PLAYER) {
+        if ((this->skelAnime.animation != &gTingleWaitAnim) && (this->skelAnime.animation != &gTingleSpinAnim)) {
+            Actor_TrackPlayer(play, &this->picto.actor, &this->headRot, &this->torsoRot, this->picto.actor.focus.pos);
+        } else {
+            Actor_TrackNone(&this->headRot, &this->torsoRot);
+        }
+    } else if (this->watchTarget == EN_BAL_WATCHTRG_FAIRY) {
+        Vec3f tatlPos;
+        player = GET_PLAYER(play);
```

Comment:
```suggestion
        Vec3f tatlPos;

        player = GET_PLAYER(play);
```
Spacing after variable declarations